0.18: Groups view updates

nucleogenesis opened the pull request:
<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

## Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->

https://github.com/user-attachments/assets/da845e9c-6a11-4dfc-8166-a5495f82aa18

- Updates strings to match new designs per #12678 
- "Learn more" link opens new "About groups" modal 
- Adds "Learn more" to common strings
- Adds "header" slot to "PlanHeader" component to permit more flexible display of title/className/description/etc
- Adds "Enroll learners" option in the "GroupsPage/TableRow" component, emitting "enroll" when selected (handling not implemented yet)

## References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->

Closes #12678 

## Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->

- [ ] Review the designs in #12678 and look at the Groups tab; see that changes indicated there are in place.
- [ ] Can rename and delete groups from new context menu on manage groups page (where you see users in the group)
- [ ] See "Enroll learners" option in each group's options dropdown in the groups table, clicking it takes you to enroll learners in that row's group
- [ ] Other tabs still work (Lessons, Quizzes)

nucleogenesis committed:
add rename group and enroll learners to common coach strings

nucleogenesis committed:
GroupsPage/GroupRow use updated strings for options; adds enroll learners option

nucleogenesis committed:
commonCoreStrings add learnMoreAction "Learn more"

This is a string used in various places, which I noticed while needing
to add it to address #12678 and not finding a common definition for the
message.

nucleogenesis committed:
coach constants: add "ABOUT_GROUP" to GroupModals

nucleogenesis committed:
PlanHeader: add a #header slot, allow override of title/description

This permits me to add the new description for the plan page and also
gives makes it easy enough to include the link/button Groups has for
"Learn more" next to the description text

nucleogenesis committed:
GroupsPage: Adds "About groups" modal, $trs & handling

nucleogenesis committed:
GroupsPage: incl class name w/ classes icon in PlanHeader#header slot

nucleogenesis committed:
GroupMembersPage: Add options menu for rename & delete group

nucleogenesis committed:
GroupsPage: handle enroll event from GroupRowTr

nucleogenesis commented on kolibri/core/assets/src/mixins/commonCoreStrings.js:
```diff
@@ -17,6 +17,11 @@ export const coreStrings = createTranslator('CommonCoreStrings', {
     context:
       'Button to cancel an action and return to the previous page. Usually this is the opposite of the save button which saves some piece of information.',
   },
+  learnMoreAction: {
```
Also defined in
- MissingresourceAlert (kolibri-common)
- DataPage (facility)
- ProfilePage/index (user_profile)

Could update those in this PR, make a follow-up issue, or just revert this change and add this same string to the GroupsPage component

pcenov commented:
Hi @nucleogenesis,

I was able to identify the following issues, which I can also report as follow-ups:

1. When I am within a group and I select the `Delete group` option then I see some errors in the console and remain at the same page seeing "This group doesn't exist". For this scenario I would expect to be brought back to the Groups page while seeing the snackbar notification that the group is deleted:

https://github.com/user-attachments/assets/19982907-26d1-4670-8d89-ff97e6466c11

2. In https://github.com/learningequality/kolibri/issues/12678 there's an image suggesting that while viewing a group one should be able to see the name of the class, while currently I can see the number of learners:

![missing class name](https://github.com/user-attachments/assets/5d57c5bb-8252-4789-8558-21594793214c)

3. When I enroll all of the available learners in a group and then open again the "Enroll learners into..." page then I see the following text `All users are already enrolled in this class`. This is not caused by changes made here, but if possible we should change it to `All users are already enrolled in this group`

![enrolled in group](https://github.com/user-attachments/assets/053fea78-a5f3-4792-b977-5c7c27f2ce0c)

nucleogenesis commented:
Thanks @pcenov your **2)** item I will address here.

For **1)** and **3)** these are great feedback. I also found it odd that there was even a "this group doesn't exist" message there rather than sending us back to the groups page.

As for changing that string, I can update it quickly here unless @marcellamaki thinks this should be a part of some upcoming strings/i18n review

marcellamaki commented:
@nucleogenesis for 3) are we currently using the "already enrolled in this class".... as an "error" string for class enrollments? might need to be an additional string, rather than a change?

nucleogenesis committed:
GroupMembersPage: Swap classname icon text for # learners message

nucleogenesis committed:
commonCoachStrings: remove `numberOfLearners` as it's only usage has been removed

see 805dc341 for last usage removed

nucleogenesis committed:
GropuEnrollPage: "All users already enrolled in this ~class~ group"

This string is used only for referencing this page which is about
enrolling learners in a particular group, not the class

nucleogenesis committed:
GroupMembersPage: After group deleted, redirect to groups page

nucleogenesis commented:
@pcenov I went to fix (2) but ended up fixing all of them.

@marcellamaki that "All learners..." is only used on that component, so I've updated it from `class` to `group` accordingly.

Additionally, I've added a line that returns the user to the Groups page if they delete the group from that Group's member list page.

nucleogenesis commented on kolibri/plugins/coach/assets/src/views/plan/GroupMembersPage/index.vue:
```diff
@@ -22,11 +22,12 @@
-          >
-            {{ coachString('numberOfLearners', { value: currentGroup.users.length }) }}
+          <KFixedGridItem span="1">
+            <KIcon
+              icon="classes"
+              class="class-name-icon"
+            />
```
I did not use a KLabeledIcon here because the classes icon is, by default, far too small compared to the text.

pcenov reviewed (approved):
Thanks @nucleogenesis - I confirm that the reported issues are fixed now - everything is implemented as specified in https://github.com/learningequality/kolibri/issues/12678

@radinamatic

marcellamaki commented on kolibri/plugins/coach/assets/src/views/plan/GroupsPage/GroupRow.vue:
```diff
@@ -49,18 +49,31 @@
+          default:
+            // eslint-disable-next-line no-console
+            console.warn(
+              `GroupRow: Tried to handleSelection of ${selectedOption}, but that isn't being handled.`,
```
> that isn't being handled

totally a nit and not blocking, but, this means there's no matching case, right? I think if you were to say that, instead of "isn't being handled", it would be clearer

marcellamaki reviewed (approved):
Thanks @nucleogenesis, looks great. I added one tiny non-blocking suggestion for comment phrasing, but fine to leave as-is if you think it's clearer
